src/screens/toDoList/ToDoList.tsx

```diff
@@ -0,0 +1,61 @@
+import React, {useMemo} from 'react';
+import {FlatList, Modal, StyleSheet, Text, View} from 'react-native';
+import ToDoListRow from './components/ToDoListRow';
+import FloatingButton from '../../ui/FloatingButton';
+import {useToDoList} from './useToDoList';
+import EditModal from '../../widgets/modal/EditModal';
+import {useModal} from '../../widgets/modal/useModal';
+
+export default function ToDoList() {
+  const {
+    toDoListArray,
+    handleAddItem,
+    handleDeleteItem,
+    openRow,
+    handleRowSwipe,
+  } = useToDoList();
+
+  const {modalVisible, openModal, closeModal} = useModal();
+
+  const buttonState = useMemo(() => {
+    return modalVisible ? '-' : '+';
+  }, [modalVisible]);
```

Comment:
Is there a reason for it to be outside useModal?

Reply:
this button is related to the floating button title ( - in case of the modal is open, + in case of the modal is closed)